supervisor/dbus/agent/__init__.py

```diff
@@ -70,6 +70,7 @@ def diagnostics(self) -> bool:
         return self.properties[DBUS_ATTR_DIAGNOSTICS]
 
     @diagnostics.setter
+    @dbus_connected
```

Comment:
let's use `dbus_property` and make sure it works for this case too. We don't need raise an exception in that case like this do